backend/oasst_backend/prompt_repository.py

```diff
@@ -314,3 +314,17 @@ def insert_reaction(self, post_id: UUID, payload: db_payload.ReactionPayload) ->
         self.db.commit()
         self.db.refresh(reaction)
         return reaction
+
+    def store_text_labels(self, text_labels: protocol_schema.TextLabels) -> TextLabels:
+        model = TextLabels(
+            api_client_id=self.api_client.id,
+            text=text_labels.text,
+            labels=text_labels.labels,
+        )
+        if text_labels.has_post_id:
+            self.fetch_post_by_frontend_post_id(text_labels.post_id, fail_if_missing=True)
+            model.post_id = text_labels.post_id
```

Comment:
What are we doing with text_labels that have no post_id ... to what do they refer? Maybe we should require either a `post_id` or a `frontend_post_id`?  Maybe you had something else in mind?

Reply:
the idea was that this is more lightweight, i.e. the text is always sent, and the table ends up being a text -> labels mapping. If the post id exists, then good, but could also be that the user "reports" just a part of a message, or something else entirely.

Reply:
I see. In general self-contained packages are good IMO ... but here we need a use-case how the data that enters the db could be used later (either manually or automatic). For automatic case it could be a sample for a dataset to train these classifications. But manual moderation (e.g. via fulltext search?) without references is a bit tricky.

Reply:
true, my thought was mainly around training the classifiers. for the moderation bit, I think it's easy to filter where post_id is not null. the question is whether we require it or not. I can definitely imagine situation where the labels aren't really matching exactly one post, for example, assume we ask the user to rank 3 assistant answers, so our discord bot makes one discord post with all 3 answers in it. the user just wants to label answer number 1 as profane. how should they do that?

Reply:
How do you envision users to submit a text labels without a post-id (e.g. where would they enter the classification, where would the text sent with the label-input come from)?

Reply:
on the website: highlight a piece of text with the mouse (or have the UI assist you). on discord, maybe there's a button (or emoji) for "label this text", after which some sort of interactive dialogue opens. yea I admit, it's probably not the most user friendly solution